pkg/daemon/ceph/osd/daemon.go

```diff
@@ -214,6 +214,14 @@ func Provision(context *clusterd.Context, agent *OsdAgent, crushLocation string)
 		return errors.Wrap(err, "failed to configure devices")
 	}
 
+	// Let's fail if no OSDs were configured
+	// This likely means the filter for available devices passed (in PVC case)
+	// but the resulting device was already configured for another cluster (disk not wiped and leftover)
+	// So we need to make sure the list is filled up, otherwise fail
+	if len(deviceOSDs) == 0 {
+		return errors.Wrap(err, "failed to configure devices or list existing osds, osd list is empty")
```

Comment:
Rather than fail, how about just logging the error? It might be expected that some nodes don't have devices and they can just be skipped for osd provisioning.

Reply:
So log and return right?

Reply:
sounds good

Reply:
done